scipy/spatial/transform/rotation.py

```diff
@@ -657,3 +657,105 @@ def __mul__(self, other):
+        `N` rotations to perform the interpolation between. At least 2
+        rotations must be specified.
+
+    Methods
+    -------
+    __call__
+
+    References
+    ----------
+    .. [1] `Quaternion Slerp
+            <https://en.wikipedia.org/wiki/Slerp#Quaternion_Slerp>`_
+    """
+    def __init__(self, t, r):
+        # We don't need to store r, just r[:-1] and interpolation rotvecs
+        if r.num_rots == 1:
+            raise ValueError("Expected at least 2 rotations in object, "
+                             "got {}.".format(r.num_rots))
+
+        t = np.asarray(t)
+        if t.ndim != 1:
+            raise ValueError("Expected times to be specified in a 1 "
+                             "dimensional array, got {} "
+                             "dimensions.".format(t.ndim))
+
+        if t.shape[0] != r.num_rots:
+            raise ValueError("Expected number of rotations to be equal to "
+                             "number of timestamps given, got {} rotations "
+                             "and {} timestamps.".format(
+                                r.num_rots, t.shape[0]))
+        self.times = t
+        self.timedelta = np.diff(t)
+
+        if np.any(self.timedelta <= 0):
+            raise ValueError("Times must be in strictly increasing order.")
+
+        # inv() and __getitem__ return new objects anyway
+        self.rots = r[:-1]
+        # should be equal to r.rots - 1
+        self.num_rots = self.rots.num_rots
+        rot_params = self.rots.inv() * r[1:]
+        self.interp_rotvecs = rot_params.as_rotvec()
+
+    def __call__(self, t):
+        """Compute interpolated rotations at given times.
+
+        Returns a `Rotation` instance with the interpolated rotations.
+
+        Parameters
+        ----------
+        t : array_like, 1D
+            Times to compute the interpolations at.
+        """
+        t = np.asarray(t)
+        if t.ndim != 1:
+            raise ValueError("Expected times to be specified in a 1 "
+                             "dimensional array, got {} "
+                             "dimensions.".format(t.ndim))
+
+        # Index of largest keyframe times less than or equal to given time.
+        # side='right' or side='left' will exclude one of the endpoints
+        # of the time range. 'left' excludes t_min.
+        ind = np.searchsorted(self.times, t) - 1
+        # Include t_min. Without this step, index for t_min equals -1
+        ind[t == self.times[0]] = 0
+        # self.num_rots = number of rotations in self.rots, not number of
+        # rotations in original object
+        if np.any(np.logical_or(ind < 0, ind > self.num_rots - 1)):
+            raise ValueError("Interpolation times must be within the range "
+                             "[{}, {}], both inclusive.".format(
+                                self.times[0], self.times[-1]))
+
+        rots = self.rots[ind]
+        interp_exponents = (t - self.times[ind]) / self.timedelta[ind]
```

Comment:
What `exponents` mean here?

Reply:
In the slerp formula, this parameter is the exponent of  `q0^(-1) * q1`. I'll change this name.